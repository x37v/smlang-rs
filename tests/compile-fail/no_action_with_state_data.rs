extern crate smlang;

use smlang::statemachine;

#[derive(PartialEq)]
pub enum Events {
    Event1,
}

pub struct Context;

statemachine! {
    transitions: {
        *State1 + Event1 = State2(u32), //~ This state has data associated, but not action is define here to provide it.
    }
}

fn main() {}
