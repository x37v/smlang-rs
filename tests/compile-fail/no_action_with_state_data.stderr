error[E0423]: expected value, found builtin type `u32`
  --> tests/compile-fail/no_action_with_state_data.rs:14:35
   |
14 |         *State1 + Event1 = State2(u32), //~ This state has data associated, but not action is define here to provide it.
   |                                   ^^^ not a value

error[E0618]: expected function, found `States`
  --> tests/compile-fail/no_action_with_state_data.rs:12:1
   |
12 | / statemachine! {
13 | |     transitions: {
14 | |         *State1 + Event1 = State2(u32), //~ This state has data associated, but not action is define here to provide it.
   | |                            ------ `States::State2` defined here
15 | |     }
16 | | }
   | |_^ call expression requires function
   |
   = note: this error originates in the macro `statemachine` (in Nightly builds, run with -Z macro-backtrace for more info)
